document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/service/OfficerService.java

```diff
@@ -34,11 +34,19 @@ public OfficersApi getOfficers(String companyNumber) throws ServiceException {
 
         officersApi = retrieveOfficerAppointments(companyNumber, officersApi, apiClient, startIndex, itemsPerPage);
         
-        while (officersApi.getItems().size() < officersApi.getTotalResults()) {
-        	startIndex += itemsPerPage;
-            OfficersApi moreResults = retrieveOfficerAppointments(companyNumber, officersApi, apiClient, startIndex, itemsPerPage);
-            officersApi.getItems().addAll(moreResults.getItems());
-        }
+		while (officersApi.getItems().size() < officersApi.getTotalResults()) {
+			try {
+				startIndex += itemsPerPage;
+				OfficersApi moreResults = retrieveOfficerAppointments(companyNumber, officersApi, apiClient, startIndex, itemsPerPage);
+				officersApi.getItems().addAll(moreResults.getItems());
+			} catch (ServiceException se) {
+				if (officersApi.getItems().size() > 0) {
```

Comment:
Are we going to log a specific warning or error here?

Reply:
We will, but I wanted to double check it works in cidev before adding logging and unit tests